src/app/courses/step-view-courses/courses-step-view.component.ts

```diff
@@ -114,4 +132,14 @@ export class CoursesStepViewComponent implements OnInit, OnDestroy {
     this.router.navigate([ 'exam', { questionNum: this.examStart } ], { relativeTo: this.route });
   }
 
+  setupList(resourcesRes) {
+    resourcesRes.forEach((res: any) => { this.resourceIds.push(res._id); });
+    this.course.steps.map(step => {
```

Comment:
Since this is a view of one step, we don't need to run this for every step on the course